pilot/pkg/model/jwks_resolver.go

```diff
@@ -224,16 +223,17 @@ func newJwksResolverWithCABundlePaths(
 
 var errEmptyPubKeyFoundInCache = errors.New("empty public key found in cache")
 
-// GetPublicKey returns the  JWT public key if it is available in the cache
+// GetPublicKey returns the JWT public key if it is available in the cache
 // or fetch with from jwksuri if there is a error while fetching then it adds the
 // jwksURI in the cache to fetch the public key in the background process
-func (r *JwksResolver) GetPublicKey(issuer string, jwksURI string) (string, error) {
+func (r *JwksResolver) GetPublicKey(issuer string, jwksURI string, timeout time.Duration) (string, error) {
 	now := time.Now()
 	key := jwtKey{issuer: issuer, jwksURI: jwksURI}
 	if val, found := r.keyEntries.Load(key); found {
 		e := val.(jwtPubKeyEntry)
 		// Update cached key's last used time.
 		e.lastUsedTime = now
+		e.timeout = timeout
```

Comment:
nit: `timeout` needs to be part of the cache key (jwtKey struct), otherwise modifying the timeout may not take effect(it is used in the JwksResolver.refresh method) before the cache expires (default is 7 days).

Reply:
Thank you for the comment. I will take a look today :)

Reply:
In the current logic the timeout is still getting updated for the next refresh which may happen after [0,20] minutes or after [0,1] minute in case the previous refresh was already failing.

Assuming the user updated the timeout because the request were already failing, then it needs up to one extra minute for the cache to catch up and start to work again, in the meantime the old cached key will still be present and be used.
I wonder that maybe it is better to keep the logic as it is without complicating it for little gain